File: 2_AddTwoNumbers/c++/AddTwoNumbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *res = new ListNode(0);
        ListNode *meta = res;
        int carry = 0;
        while((l1!=NULL)||(l2!=NULL)||(carry!=0)){
            meta->next = new ListNode(0);
            meta = meta->next;
            int mSum = ((l1==NULL?0:l1->val)+(l2==NULL?0:l2->val)+carry);
            meta->val = mSum%10;
            carry = mSum/10;
            l1 = (l1==NULL?NULL:l1->next);
            l2 = (l2==NULL?NULL:l2->next);
            
            
        }
        return res->next;
    }
};

File: 2_AddTwoNumbers/python/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: 'ListNode', l2: 'ListNode') -> 'ListNode':
        res_head = ListNode(0)
        meta = res_head
        carry = 0
        while (l1 != None) or (l2 != None) or (carry != 0):
            meta.next = ListNode(0)
            meta = meta.next
            mSum = carry
            if l1 != None:
                mSum += l1.val
                l1 = l1.next
            if l2 != None:
                mSum += l2.val
                l2 = l2.next
            meta.val = mSum%10
            carry = mSum//10
            
        return res_head.next
        